sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3296,6 +3296,18 @@ SIREPO.app.directive('ldapLogin', function (requestSender) {
 });
 
 SIREPO.app.directive('commonFooter', function() {
+    const _refreshModals = () => {
+        return Object.keys(SIREPO.refreshModalMap).reduce(_refreshTemplate, '');
+    };
+
+    const _refreshTemplate = (rv, id) => {
+        const x = SIREPO.refreshModalMap[id];
+        if (! x) {
+            throw new Error(`not found id=${id} in refreshModalMap`);
+        }
+        return rv + `<div data-confirmation-modal="" data-is-required="true" data-id="${x.modal}" data-title="${x.title}" data-ok-text="Refresh" data-ok-clicked="refreshPage()">${x.msg}. Select <b>Refresh</b> to update this simulation.</div>\n`;
+    };
```

Comment:
I think a map() rather than reduce() is simpler here:
```javascript
    const _refreshModals = () => {
        return Object.values(SIREPO.refreshModalMap).map(_refreshTemplate).join('');
    };

    const _refreshTemplate = (x) => {
        return `<div data-confirmation-modal="" data-is-required="true" data-id="${x.modal}" data-title="${x.title}" data-ok-text="Refresh" data-ok-clicked="refreshPage()">${x.msg}. Select <b>Refresh</b> to update this simulation.</div>\n`;
    };
```

Reply:
`map().join('')` is exactly the same as `reduce('')` so I think one function call is simpler than two.

Reply:
Rather than keys(), If you call ```Object.values(SIREPO.refreshModalMap).reduce(_refreshTemplate, '')``` then you won't need the error handling in _refreshTemplate()

Reply:
Thanks! [cfc5fc9](https://github.com/radiasoft/sirepo/pull/6751/commits/cfc5fc9bf9751306ddf38f8009d6ad53a523b116)